File: RGR/app/src/main/java/com/example/rgr/Comparator/ComparatorInteger.java
package com.example.rgr.Comparator;

import com.example.rgr.Builder.BuilderInteger;

import java.io.Serializable;


public class ComparatorInteger implements Comparator, Serializable
{
    @Override
    public int compare(Object o1, Object o2) {
        if (((BuilderInteger) o1).value == ((BuilderInteger) o2).value) return 0;
        if (((BuilderInteger) o1).value > ((BuilderInteger) o2).value) return 1;
        else return -1;
    }
}

File: LR_3/src/git/group/Comparator/ComparatorInteger.kt
package git.group.Comparator

import git.group.Builder.BuilderInteger
import java.io.Serializable

class ComparatorInteger : Comparator, Serializable {
    override fun compare(o1: Any?, o2: Any?): Int {
        if ((o1 as BuilderInteger?)!!.value == (o2 as BuilderInteger?)!!.value) return 0
        return if (o1!!.value > o2!!.value) 1 else -1
    }
}